src/tasks/AotCompilerTask/MonoAOTCompiler.csproj

```diff
@@ -11,10 +11,14 @@
     <NoWarn Condition="$(TargetFramework.StartsWith('net4'))">$(NoWarn),CS8604,CS8602</NoWarn>
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
   </PropertyGroup>
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppToolCurrent)'">
     <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildVersion)" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" />
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetFrameworkToolCurrent)'">
+    <!-- On .NET Framework, make sure we don't include a copy of the MSBuild assemblies with the task. The NETCore version doesn't do it already. -->
+    <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildVersion)" IncludeAssets="compile" />
+    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" IncludeAssets="compile" />
   </ItemGroup>
```

Comment:
This pattern looks a bit verbose to me. Why use `IncludeAssets=compile` instead of `ExcludeAssets=runtime`? We usually do the latter, i.e. in https://github.com/dotnet/arcade/blob/d9fefd08b4012031f9e05852a0a85ba50bbb8f7d/eng/BuildTask.targets#L78

I think we should do "globally" for all task projects in the src/tasks Directory.Build.props file similar to how this works in Arcade.

Reply:
I don't know, I just copied what we already had :)
I agree that unifying to the arcade model sounds better.

Reply:
>  e.g. ILCompiler.Build.Tasks.csproj and ILLink.Tasks.csproj are in other paths.

I think we should be able to move them to `src/tasks`.

Reply:
They shouldn't be moved to src/tasks. That directory is restored and built differently than the rest of the product in dotnet/runtime. Reason for that is that src/tasks was meant to only store non-shipping repo tasks that are needed inside the repository. It currently doesn't share the same msbuild node as the rest of the repo to avoid task lock issues, cause we don't disable nodereuse in runtime.

I would rather prefer to introduce infrastructure that allows to build tasks with these VS constraints from any location in the repo. Ideally any shipping task would eventually be moved out of src/tasks. And for the shipping tasks that are also invoked in-tree, they could leverage the new short-lived taskhostfactory msbuild feature to avoid the file locks.